river/Cursor.zig

```diff
@@ -747,26 +786,99 @@ fn processMotion(self: *Self, device: *wlr.InputDevice, time: u32, delta_x: f64,
         },
         .resize => |data| {
             const border_width = if (data.view.draw_borders) server.config.border_width else 0;
-
-            // Set width/height of view, clamp to view size constraints and output dimensions
             const box = &data.view.pending.box;
-            box.width = @intCast(u32, math.max(0, @intCast(i32, box.width) + @floatToInt(i32, delta_x)));
-            box.height = @intCast(u32, math.max(0, @intCast(i32, box.height) + @floatToInt(i32, delta_y)));
+            const constraints = data.view.getConstraints();
+            const min_width = @intCast(i32, math.max(View.min_size, constraints.min_width));
+            const min_height = @intCast(i32, math.max(View.min_size, constraints.min_height));
+
+            // Using two points instead of a box makes this logic a bit more
+            // readable for humans. The compiler should optimize the conversion out.
+            var botright_x = box.x + @intCast(i32, box.width);
+            var botright_y = box.y + @intCast(i32, box.height);
+
+            switch (data.corner) {
+                .topleft => {
+                    // If the max_{width,height} is larger than the max of u16,
+                    // it is larger than any existing screen anyway and usually
+                    // means that the view has not maximum size, so we can
+                    // truncate without losing relevant data.
```

Comment:
We could truncate to a u31 instead, and use `@as()` instead of `@intCast()` after truncation.

Reply:
Interestingly, `@truncate()`-ing to u31 crashes.